### DISCUSSION TITLE
Webview2 not able to upload stream more than 65MB stream in "CoreWebView2.Environment.CreateWebResourceResponse(stream, 200, "OK", string.Empty);"

### DISCUSSION THREAD

**Discussion by SajidMemon44** [MicrosoftEdge/WebView2Feedback#4525] [Q&A]
Hi,

I am using webview2 , i want to send file using "CoreWebView2.Environment.CreateWebResourceResponse(stream, 200, "OK", string.Empty);" file size is more than 70MB it was cause error 

   at MS.Win32.UnsafeNativeMethods.DispatchMessage(MSG& msg)
   at System.Windows.Threading.Dispatcher.PushFrameImpl(DispatcherFrame frame)
   at System.Windows.Threading.Dispatcher.PushFrame(DispatcherFrame frame)
   at System.Windows.Application.RunDispatcher(Object ignore)
   at System.Windows.Application.RunInternal(Window window)
   at System.Windows.Application.Run(Window window)
   at System.Windows.Application.Run()
   at WinProjectE.App.Main()
   
   System.Runtime.InteropServices.SEHException
  HResult=0x80004005
  Message=External component has thrown an exception.
  Source=<Cannot evaluate the exception source>
  StackTrace:
<Cannot evaluate the exception stack trace>

Method used :

private void CoreWebView2_WebResourceRequested(object sender, CoreWebView2WebResourceRequestedEventArgs e)
{
       if (e.Request.Uri.Contains("/XXX/XXX/XXX/XXX/", StringComparison.OrdinalIgnoreCase))
        {
                    string filename = Path.GetFileName(e.Request.Uri);
                     string filePath = Path.Combine(App.oGlobalInput.TempDirectory, filename);
        
                    if (File.Exists(filePath))
                     {
                                FileStream stream = File.OpenRead(filePath);
                                e.Response = parent.webView.CoreWebView2.Environment.CreateWebResourceResponse(stream, 200, "OK", 
                                   string.Empty);
                   }
                    else
                    {
                                e.Response =
                                   parent.webView.CoreWebView2.Environment.CreateWebResourceResponse(null, 404, "File not found", string.Empty);
                    }
          }
 }
 
 We try this approach as well but still same error : **https://learn.microsoft.com/en-us/microsoft-edge/webview2/concepts/working-with-local-content?tabs=dotnetcsharp**

![Webview2Error](https://github.com/MicrosoftEdge/WebView2Feedback/assets/168806411/a04f9cbe-c209-40e7-9232-4a23833f52dc)

**Comment by mdCllang** [comment]
We run in the exact the same issue as you descripe.
Some additional Infos:

We use WPF, NetFramework 4.8, 
Nuget: Microsoft.Web.WebView2: Versions tested: 1.0.1823.32 and 1.0.2151.40 (latest at moment of posting)

TestPC runs on Windows 11:
WebView2 Runtime: 128.0.2739.79 (latest at moment of posting)

Any suggestions or ideas how to fix it would be nice.

**Reply by brucewedding** [reply]
Try giving it a content type and length. The size trigger |(>65MB) is a clue that you're overrunning the buffer. If you give it the size, it will allocate more memory.

```private void CoreWebView2_WebResourceRequested(object sender, CoreWebView2WebResourceRequestedEventArgs e)
{
    if (e.Request.Uri.Contains("/XXX/XXX/XXX/XXX/", StringComparison.OrdinalIgnoreCase))
    {
        string filename = Path.GetFileName(e.Request.Uri);
        string filePath = Path.Combine(App.oGlobalInput.TempDirectory, filename);

        if (File.Exists(filePath))
        {
            try
            {
                // Get file info to determine size and mime type
                FileInfo fileInfo = new FileInfo(filePath);
                string contentType = GetMimeTypeFromFileName(filename);

                // Create response headers with content length and type
                var headers = $"Content-Type: {contentType}\r\nContent-Length: {fileInfo.Length}";

                // Open file as stream
                FileStream stream = File.OpenRead(filePath);

                // Create response with headers
                e.Response = parent.webView.CoreWebView2.Environment.CreateWebResourceResponse(
                    stream, 200, "OK", headers);

                // Register for the Completed event to dispose the stream
                e.GetDeferral().Complete();
            }
            catch (Exception ex)
            {
                // Log the error for debugging
                System.Diagnostics.Debug.WriteLine($"Error serving file: {ex.Message}");
                e.Response = parent.webView.CoreWebView2.Environment.CreateWebResourceResponse(
                    null, 500, "Internal Server Error", string.Empty);
            }
        }
        else
        {
            e.Response = parent.webView.CoreWebView2.Environment.CreateWebResourceResponse(
                null, 404, "File not found", string.Empty);
        }
    }
}

private string GetMimeTypeFromFileName(string fileName)
{
    string extension = Path.GetExtension(fileName).ToLowerInvariant();

    // Add more mappings as needed
    switch (extension)
    {
        case ".pdf": return "application/pdf";
        case ".jpg":
        case ".jpeg": return "image/jpeg";
        case ".png": return "image/png";
        case ".mp4": return "video/mp4";
        case ".zip": return "application/zip";
        case ".txt": return "text/plain";
        case ".csv": return "text/csv";
        case ".xlsx": return "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet";
        // Add other types as needed
        default: return "application/octet-stream";
    }
}```
